From: Scott Leggett <scott@sl.id.au>
Date: Sun, 5 Nov 2017 22:03:17 +1100
Subject: Fix typo in dump-bgp help text.

---
 bgpd/bgp_dump.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bgpd/bgp_dump.c b/bgpd/bgp_dump.c
index 01f9b41..2b8513c 100644
--- a/bgpd/bgp_dump.c
+++ b/bgpd/bgp_dump.c
@@ -741,8 +741,8 @@ DEFUN (dump_bgp_all,
        "dump bgp (all|all-et|updates|updates-et|routes-mrt) PATH [INTERVAL]",
        "Dump packet\n"
        "BGP packet dump\n"
-       "Dump all BGP packets\nDump all BGP packets (Extended Tiemstamp Header)\n"
-       "Dump BGP updates only\nDump BGP updates only (Extended Tiemstamp Header)\n"
+       "Dump all BGP packets\nDump all BGP packets (Extended Timestamp Header)\n"
+       "Dump BGP updates only\nDump BGP updates only (Extended Timestamp Header)\n"
        "Dump whole BGP routing table\n"
        "Output filename\n"
        "Interval of output\n")
